tests/foreman/api/test_operatingsystem.py

```diff
@@ -50,15 +53,101 @@ def test_bz_1114640(self):
 class OSTestCase(APITestCase):
     """Tests for operating systems."""
 
+    @classmethod
+    def setUpClass(cls):
+        """Create an organization which can be re-used in tests."""
+        super(OSTestCase, cls).setUpClass()
+        cls.org = entities.Organization().create()
+
+    @run_only_on('sat')
+    def test_positive_create_different_names(self):
+        """@Test: Create operating system with valid name only
+
+        @Feature: Operating System
+
+        @Assert: Operating system entity is created and has proper name
+        """
+        for name in valid_data_list():
+            with self.subTest(name):
+                os = entities.OperatingSystem(name=name).create()
+                self.assertEqual(os.name, name)
+
     @run_only_on('sat')
-    def test_point_to_arch(self):
+    def test_positive_create_different_os_family(self):
+        """@Test: Create operating system with every OS family possible
+
+        @Feature: Operating System
+
+        @Assert: Operating system entity is created and has proper OS family
+        assigned
+        """
+        for os_family in OPERATING_SYSTEMS:
+            with self.subTest(os_family):
+                os = entities.OperatingSystem(family=os_family).create()
+                self.assertEqual(os.family, os_family)
+
+    @run_only_on('sat')
+    def test_positive_create_with_minor(self):
+        """@Test: Create operating system with minor version
+
+        @Feature: Operating System
+
+        @Assert: Operating system entity is created and has proper minor
+        version
+        """
+        minor_version = gen_string('numeric')
+        os = entities.OperatingSystem(minor=minor_version).create()
+        self.assertEqual(os.minor, minor_version)
+
+    @run_only_on('sat')
+    @skip_if_bug_open('bugzilla', 1230902)
+    def test_create_with_minor_1230902(self):
+        """@Test: Create an operating system with an integer minor version.
+
+        @Feature: Operating System
+
+        @Assert: The minor version can be read back as a string.
+        """
+        minor = gen_integer(0, 9999999999999999)  # max len: 16 chars
```

Comment:
1) It is not mine code, so I don'y know what was initial intention
2) I don't like that gen_string('numeric') can generate values like '00001' which will be cut out by further transformation.
3) '999999...' is not hardcoded value, but just max value that can be used, so no problem here. Maybe it doesn't look so nice from aesthetic point of view, but anyway.

However, no serious reasons to not make proposed change :)
Fixed.